### PR TITLE
[BSD] Restore support

### LINKED ISSUE #637: [BSD] Pass tests with KqueueEmitter
Discovered while working on #632 which made watchdog not work on BSD systems.

By separating in two issues we can fix things progressively.

8 tests fail currently:

- [x] `tests/test_emitter.py::test_create FAILED`
- [x] `tests/test_emitter.py::test_move_to FAILED`
- [x] `tests/test_emitter.py::test_separate_consecutive_moves FAILED`
- [x] `tests/test_emitter.py::test_fast_subdirectory_creation_deletion FAILED`
- [x] `tests/test_emitter.py::test_recursive_on FAILED`
- [x] `tests/test_emitter.py::test_recursive_off FAILED`
- [x] `tests/test_emitter.py::test_renaming_top_level_directory FAILED`
- [x] `tests/test_emitter.py::test_move_nested_subdirectories FAILED`

### PR DESCRIPTION
**Pull request by evilham**
See: https://cirrus-ci.com/task/5908114574671872
Fixes #637

For some reason GH is not taking into account that some commits are merged already? Maybe some cache issue...

### CODE REVIEW

**Comment by BoboTiG** [comment]
You need to rebase your branch on master before adding the new commit.

File: src/watchdog/observers/kqueue.py [hunk 1]
```diff
@@ -672,33 +659,33 @@ def queue_events(self, timeout):
...
-                    for src_path in dirs_renamed:
-                        self._queue_renamed(src_path,
-                                            True,
-                                            ref_snapshot,
-                                            new_snapshot)
-                    self._queue_dirs_modified(dirs_modified,
-                                              ref_snapshot,
-                                              new_snapshot)
             except OSError as e:
                 if e.errno == errno.EBADF:
```
Comment (BoboTiG) [inline src/watchdog/observers/kqueue.py]:
```suggestion
                if e.errno != errno.EBADF:
```

File: src/watchdog/observers/kqueue.py [hunk 2]
```diff
@@ -672,33 +659,33 @@ def queue_events(self, timeout):
...
-                    # logging.debug(e)
                     pass
```
Comment (BoboTiG) [inline src/watchdog/observers/kqueue.py]:
```suggestion
                    raise
```

File: src/watchdog/observers/kqueue.py [hunk 3]
```diff
@@ -672,33 +659,33 @@ def queue_events(self, timeout):
...
                 else:
```
Comment (BoboTiG) [inline src/watchdog/observers/kqueue.py]:
```suggestion

```

File: src/watchdog/observers/kqueue.py [hunk 4]
```diff
@@ -672,33 +659,33 @@ def queue_events(self, timeout):
...
                     raise
```
Comment (BoboTiG) [inline src/watchdog/observers/kqueue.py]:
```suggestion

```

**Comment by BoboTiG** [comment]
This is awesome, many thanks @evilham !!

**Comment by BoboTiG** [comment]
In the Cirrus-CI report, there is those lines at the end:
```python
Exception in thread Thread-8:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/tmp/cirrus-ci-build/src/watchdog/observers/api.py", line 145, in run
    self.queue_events(self.timeout)
  File "/tmp/cirrus-ci-build/src/watchdog/observers/kqueue.py", line 668, in queue_events
    self.watch.is_recursive)
  File "/tmp/cirrus-ci-build/src/watchdog/utils/dirsnapshot.py", line 253, in __init__
    st = self.stat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp_tebxtss/dir1'
```

Does that mean the test was retried and passes?

**Comment by evilham** [comment]
Thanks for the comment, totally missed that :-).

Re: message, yeah, I think that test is marked as flaky and it passes on a second go; didn't observe that locally.

It'd look like the directory gets removed before a new dir snapshot was created (L668); unsure if this is an error in kqueue or in dirsnapshot o.ô.
What would you think is best to handle this case?

Waiting for https://cirrus-ci.com/task/5449374754930688 to finish and check if it's flaky as well.

**Comment by BoboTiG** [comment]
If this is a random error, let's do nothing :)

**Comment by evilham** [comment]
Yup, cirrus-ci didn't replicate that either:
https://cirrus-ci.com/task/5274372285923328

**Comment by BoboTiG** [comment]
Merged, thank you for your time :champagne:

**Comment by evilham** [comment]
Party! thank you for your kind assistance :-).

**Pull request merged**